### PR TITLE
Release Spanner libraries version 3.5.0

### PR DESCRIPTION
**Pull request by jskeet**
Changes in Google.Cloud.Spanner.Data version 3.5.0:

- [Commit 65f6e7b](https://github.com/googleapis/google-cloud-dotnet/commit/65f6e7b):
  - feat: log CommitStats if requested by the client application ([issue 5506](https://github.com/googleapis/google-cloud-dotnet/issues/5506))
- [Commit 8562ced](https://github.com/googleapis/google-cloud-dotnet/commit/8562ced): fix: use null instead of empty array + integration test
- [Commit 0343ff8](https://github.com/googleapis/google-cloud-dotnet/commit/0343ff8):
  - fix: parse WITH command as SELECT. Fixes [issue 5857](https://github.com/googleapis/google-cloud-dotnet/issues/5857)
- [Commit 4866adf](https://github.com/googleapis/google-cloud-dotnet/commit/4866adf): fix: NUMERIC is missing default mapping
- [Commit a223bd0](https://github.com/googleapis/google-cloud-dotnet/commit/a223bd0): fix: local date could lead to wrong date in database

Packages in this release:
- Release Google.Cloud.Spanner.Admin.Database.V1 version 3.5.0
- Release Google.Cloud.Spanner.Admin.Instance.V1 version 3.5.0
- Release Google.Cloud.Spanner.Common.V1 version 3.5.0
- Release Google.Cloud.Spanner.Data version 3.5.0
- Release Google.Cloud.Spanner.V1 version 3.5.0

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Comment by jskeet** [comment]
(Waiting until the release process is fixed before merging.)

**Pull request merged**
